### DISCUSSION TITLE
expressLRS configurator 编译固件报错信息如何解决

### DISCUSSION THREAD

**Discussion by FPVLOVE** [ExpressLRS/ExpressLRS#1967] [General]
Warning! `src_filter` configuration option in section [env:native] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:AXIS_THOR_2400_TX_via_UART] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:AXIS_THOR_2400_RX_via_UART] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:Frsky_TX_R9M_via_STLINK] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:Frsky_TX_R9M_LITE_PRO_via_STLINK] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:Frsky_RX_R9MM_R9MINI_via_STLINK] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:Frsky_RX_R9SLIM_via_BetaflightPassthrough] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:Frsky_RX_R9SLIMPLUS_via_BetaflightPassthrough] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:Frsky_RX_R9MX_via_STLINK] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:Jumper_RX_R900MINI_via_BetaflightPassthrough] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:HappyModel_TX_ES915TX_via_STLINK] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:HappyModel_TX_ES900TX_via_UART] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:HappyModel_RX_ES915RX_via_STLINK] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:HappyModel_ES24TX_2400_TX_via_UART] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:HappyModel_ES24TX_Pro_Series_2400_TX_via_UART] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:BETAFPV_900_TX_via_UART] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:BETAFPV_900_TX_MICRO_via_UART] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:BETAFPV_900_RX_via_UART] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:BETAFPV_2400_TX_via_UART] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:BETAFPV_2400_TX_MICRO_via_UART] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:BETAFPV_2400_RX_via_UART] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:GHOST_2400_TX_via_STLINK] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:GHOST_ATTO_2400_RX_via_STLINK] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:NamimnoRC_VOYAGER_900_TX_via_STLINK] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:NamimnoRC_VOYAGER_900_OLED_TX_via_UART] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:NamimnoRC_VOYAGER_900_RX_via_STLINK] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:NamimnoRC_VOYAGER_900_ESP_RX_via_UART] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:NamimnoRC_FLASH_2400_TX_via_STLINK] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:NamimnoRC_FLASH_2400_OLED_TX_via_UART] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:NamimnoRC_FLASH_2400_RX_via_STLINK] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:NamimnoRC_FLASH_2400_ESP_RX_via_UART] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:FM30_TX_via_STLINK] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:FM30_RX_MINI_via_STLINK] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:FM30_RX_MINI_AS_TX_via_STLINK] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:HGLRC_Hermes_2400_TX_via_UART] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:QuadKopters_JR_2400_TX_via_UART] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:DIY_900_TX_TTGO_V1_SX127x_via_UART] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:DIY_900_TX_TTGO_V2_SX127x_via_UART] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:DIY_900_TX_ESP32_SX127x_E19_via_UART] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:DIY_900_TX_ESP32_SX127x_RFM95_via_UART] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:DIY_900_RX_ESP8285_SX127x_via_UART] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:DIY_900_RX_HUZZAH_RFM95W_via_UART] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:DIY_900_RX_PWMP_via_UART] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:DIY_2400_TX_ESP32_SX1280_Mini_via_UART] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:DIY_2400_TX_ESP32_SX1280_E28_via_UART] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:DIY_2400_TX_ESP32_SX1280_LORA1280F27_via_UART] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:DIY_2400_TX_ESP8285_SX1280_via_UART] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:DIY_2400_TX_DUPLETX_via_ETX] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:DIY_2400_RX_ESP8285_SX1280_via_UART] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:DIY_2400_RX_PWMP_via_UART] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:DIY_2400_RX_STM32_CCG_Nano_v0_5_via_STLINK] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:DIY_BLE_JOYSTICK_via_UART] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:MATEK_2400_RX_via_UART] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:iFlight_900_TX_via_UART] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:iFlight_2400_TX_via_UART] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:iFlight_2400_RX_via_UART] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:Vantac_Lite_2400_TX_via_UART] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:Vantac_2400_RX_via_UART] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:Jumper_AION_2400_T-Pro_TX_via_WIFI] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:Jumper_AION_NANO_2400_TX_via_UART] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:RadioMaster_Zorro_2400_TX_via_ETX] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Processing BETAFPV_2400_RX_via_UART (platform: espressif8266@3.2.0; board: esp8285-8285; framework: arduino)
--------------------------------------------------------------------------------
Verbose mode can be enabled via `-v, --verbose` option
UID bytes: 1,166,209,183,121,226
fatal: not a git repository (or any of the parent directories): .git
***Building for ESP8266/ESP8285 Platform
AUTO_WIFI_ON_INTERVAL = ON
CONFIGURATION: https://docs.platformio.org/page/boards/espressif8266/esp8285-8285.html
PLATFORM: Espressif 8266 (3.2.0) > Generic ESP8285 Module
HARDWARE: ESP8266 160MHz, 80KB RAM, 1MB Flash
PACKAGES: 
 - framework-arduinoespressif8266 @ 3.30002.0 (3.0.2) 
 - tool-esptool @ 1.413.0 (4.13) 
 - tool-esptoolpy @ 1.30000.201119 (3.0.0) 
 - toolchain-xtensa @ 2.100300.210717 (10.3.0)
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ chain, Compatibility ~ soft
Found 72 compatible libraries
Scanning dependencies...
Dependency Graph
|-- ESPAsyncWebServer-esphome @ 2.1.0
|   |-- ESPAsyncTCP-esphome @ 1.2.3
|   |-- Hash @ 1.0
|   |-- ESP8266WiFi @ 1.0
|-- SX1280Driver
|   |-- logging
|   |-- SPI @ 1.0
|-- CONFIG
|   |-- EEPROM
|   |   |-- logging
|   |   |-- EEPROM @ 1.0
|   |   |-- Wire @ 1.0
|   |-- logging
|   |-- OTA
|   |   |-- CRSF
|   |   |   |-- CRC
|   |   |   |-- CrsfProtocol
|   |   |   |   |-- CRC
|   |   |   |-- MSP
|   |   |   |   |-- logging
|   |   |   |-- TelemetryProtocol
|   |   |   |-- DEVICE
|   |   |   |   |-- helpers
|   |   |   |   |-- logging
|   |   |   |   |-- SX1280Driver
|   |   |   |   |   |-- logging
|   |   |   |   |   |-- SPI @ 1.0
|   |   |   |-- FIFO
|   |   |   |   |-- logging
|   |   |   |-- helpers
|   |   |   |-- logging
|   |-- POWERMGNT
|   |   |-- DAC
|   |   |   |-- Wire @ 1.0
|   |   |   |-- helpers
|   |   |   |-- logging
|   |   |-- DEVICE
|   |   |   |-- helpers
|   |   |   |-- logging
|   |   |   |-- SX1280Driver
|   |   |   |   |-- logging
|   |   |   |   |-- SPI @ 1.0
|   |   |-- SX1280Driver
|   |   |   |-- logging
|   |   |   |-- SPI @ 1.0
|   |   |-- helpers
|-- CRC
|-- CRSF
|   |-- CRC
|   |-- CrsfProtocol
|   |   |-- CRC
|   |-- MSP
|   |   |-- logging
|   |-- TelemetryProtocol
|   |-- DEVICE
|   |   |-- helpers
|   |   |-- logging
|   |   |-- SX1280Driver
|   |   |   |-- logging
|   |   |   |-- SPI @ 1.0
|   |-- FIFO
|   |   |-- logging
|   |-- helpers
|   |-- logging
|-- BUTTON
|   |-- DEVICE
|   |   |-- helpers
|   |   |-- logging
|   |   |-- SX1280Driver
|   |   |   |-- logging
|   |   |   |-- SPI @ 1.0
|   |-- logging
|   |-- POWERMGNT
|   |   |-- DAC
|   |   |   |-- Wire @ 1.0
|   |   |   |-- helpers
|   |   |   |-- logging
|   |   |-- DEVICE
|   |   |   |-- helpers
|   |   |   |-- logging
|   |   |   |-- SX1280Driver
|   |   |   |   |-- logging
|   |   |   |   |-- SPI @ 1.0
|   |   |-- SX1280Driver
|   |   |   |-- logging
|   |   |   |-- SPI @ 1.0
|   |   |-- helpers
|-- DEVICE
|   |-- helpers
|   |-- logging
|   |-- SX1280Driver
|   |   |-- logging
|   |   |-- SPI @ 1.0
|-- LED
|   |-- DEVICE
|   |   |-- helpers
|   |   |-- logging
|   |   |-- SX1280Driver
|   |   |   |-- logging
|   |   |   |-- SPI @ 1.0
|   |-- CrsfProtocol
|   |   |-- CRC
|   |-- POWERMGNT
|   |   |-- DAC
|   |   |   |-- Wire @ 1.0
|   |   |   |-- helpers
|   |   |   |-- logging
|   |   |-- DEVICE
|   |   |   |-- helpers
|   |   |   |-- logging
|   |   |   |-- SX1280Driver
|   |   |   |   |-- logging
|   |   |   |   |-- SPI @ 1.0
|   |   |-- SX1280Driver
|   |   |   |-- logging
|   |   |   |-- SPI @ 1.0
|   |   |-- helpers
|   |-- logging
|-- WIFI
|   |-- DEVICE
|   |   |-- helpers
|   |   |-- logging
|   |   |-- SX1280Driver
|   |   |   |-- logging
|   |   |   |-- SPI @ 1.0
|   |-- CONFIG
|   |   |-- EEPROM
|   |   |   |-- logging
|   |   |   |-- EEPROM @ 1.0
|   |   |   |-- Wire @ 1.0
|   |   |-- logging
|   |   |-- OTA
|   |   |   |-- CRSF
|   |   |   |   |-- CRC
|   |   |   |   |-- CrsfProtocol
|   |   |   |   |   |-- CRC
|   |   |   |   |-- MSP
|   |   |   |   |   |-- logging
|   |   |   |   |-- TelemetryProtocol
|   |   |   |   |-- DEVICE
|   |   |   |   |   |-- helpers
|   |   |   |   |   |-- logging
|   |   |   |   |   |-- SX1280Driver
|   |   |   |   |   |   |-- logging
|   |   |   |   |   |   |-- SPI @ 1.0
|   |   |   |   |-- FIFO
|   |   |   |   |   |-- logging
|   |   |   |   |-- helpers
|   |   |   |   |-- logging
|   |   |-- POWERMGNT
|   |   |   |-- DAC
|   |   |   |   |-- Wire @ 1.0
|   |   |   |   |-- helpers
|   |   |   |   |-- logging
|   |   |   |-- DEVICE
|   |   |   |   |-- helpers
|   |   |   |   |-- logging
|   |   |   |   |-- SX1280Driver
|   |   |   |   |   |-- logging
|   |   |   |   |   |-- SPI @ 1.0
|   |   |   |-- SX1280Driver
|   |   |   |   |-- logging
|   |   |   |   |-- SPI @ 1.0
|   |   |   |-- helpers
|   |-- helpers
|   |-- HWTIMER
|   |   |-- logging
|   |-- logging
|   |-- POWERMGNT
|   |   |-- DAC
|   |   |   |-- Wire @ 1.0
|   |   |   |-- helpers
|   |   |   |-- logging
|   |   |-- DEVICE
|   |   |   |-- helpers
|   |   |   |-- logging
|   |   |   |-- SX1280Driver
|   |   |   |   |-- logging
|   |   |   |   |-- SPI @ 1.0
|   |   |-- SX1280Driver
|   |   |   |-- logging
|   |   |   |-- SPI @ 1.0
|   |   |-- helpers
|   |-- SX1280Driver
|   |   |-- logging
|   |   |-- SPI @ 1.0
|   |-- DNSServer @ 1.1.1
|   |   |-- ESP8266WiFi @ 1.0
|   |-- ESP8266mDNS @ 1.2
|   |   |-- ESP8266WiFi @ 1.0
|   |-- ESP8266WiFi @ 1.0
|   |-- ESPAsyncWebServer-esphome @ 2.1.0
|   |   |-- ESPAsyncTCP-esphome @ 1.2.3
|   |   |-- Hash @ 1.0
|   |   |-- ESP8266WiFi @ 1.0
|-- EEPROM
|   |-- logging
|   |-- EEPROM @ 1.0
|   |-- Wire @ 1.0
|-- FHSS
|   |-- SX1280Driver
|   |   |-- logging
|   |   |-- SPI @ 1.0
|   |-- logging
|-- helpers
|-- HWTIMER
|   |-- logging
|-- logging
|-- LQCALC
|-- MSP
|   |-- logging
|-- OTA
|   |-- CRSF
|   |   |-- CRC
|   |   |-- CrsfProtocol
|   |   |   |-- CRC
|   |   |-- MSP
|   |   |   |-- logging
|   |   |-- TelemetryProtocol
|   |   |-- DEVICE
|   |   |   |-- helpers
|   |   |   |-- logging
|   |   |   |-- SX1280Driver
|   |   |   |   |-- logging
|   |   |   |   |-- SPI @ 1.0
|   |   |-- FIFO
|   |   |   |-- logging
|   |   |-- helpers
|   |   |-- logging
|-- PFD
|-- POWERMGNT
|   |-- DAC
|   |   |-- Wire @ 1.0
|   |   |-- helpers
|   |   |-- logging
|   |-- DEVICE
|   |   |-- helpers
|   |   |-- logging
|   |   |-- SX1280Driver
|   |   |   |-- logging
|   |   |   |-- SPI @ 1.0
|   |-- SX1280Driver
|   |   |-- logging
|   |   |-- SPI @ 1.0
|   |-- helpers
|-- StubbornReceiver
|-- StubbornSender
|-- Telemetry
|   |-- CrsfProtocol
|   |   |-- CRC
|   |-- CRSF
|   |   |-- CRC
|   |   |-- CrsfProtocol
|   |   |   |-- CRC
|   |   |-- MSP
|   |   |   |-- logging
|   |   |-- TelemetryProtocol
|   |   |-- DEVICE
|   |   |   |-- helpers
|   |   |   |-- logging
|   |   |   |-- SX1280Driver
|   |   |   |   |-- logging
|   |   |   |   |-- SPI @ 1.0
|   |   |-- FIFO
|   |   |   |-- logging
|   |   |-- helpers
|   |   |-- logging
|-- TelemetryProtocol
|-- Servo @ 1.0.2
Building in release mode
Installing pexpectCollecting streamexpect
  Using cached streamexpect-0.2.1.tar.gz (13 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'error'
  error: subprocess-exited-with-error
  
  �� python setup.py egg_info did not run successfully.
  �� exit code: 1
  �t��> [22 lines of output]
      C:\Users\gongxiaohui_2022\.platformio\penv\lib\site-packages\setuptools\installer.py:27: SetuptoolsDeprecationWarning: setuptools.installer is deprecated. Requirements should be satisfied by a PEP 517 installer.
        warnings.warn(
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "C:\Users\gongxiaohui_2022\AppData\Local\Temp\pip-install-lnsomkd3\streamexpect_82c3615fd898415aa39c746558caadae\setup.py", line 20, in <module>
          setup(
        File "C:\Users\gongxiaohui_2022\.platformio\penv\lib\site-packages\setuptools\__init__.py", line 87, in setup
          return distutils.core.setup(**attrs)
        File "C:\Users\gongxiaohui_2022\.platformio\penv\lib\site-packages\setuptools\_distutils\core.py", line 147, in setup
          _setup_distribution = dist = klass(attrs)
        File "C:\Users\gongxiaohui_2022\.platformio\penv\lib\site-packages\setuptools\dist.py", line 476, in __init__
          _Distribution.__init__(
        File "C:\Users\gongxiaohui_2022\.platformio\penv\lib\site-packages\setuptools\_distutils\dist.py", line 282, in __init__
          self.finalize_options()
        File "C:\Users\gongxiaohui_2022\.platformio\penv\lib\site-packages\setuptools\dist.py", line 900, in finalize_options
          ep(self)
        File "C:\Users\gongxiaohui_2022\.platformio\penv\lib\site-packages\setuptools\dist.py", line 920, in _finalize_setup_keywords
          ep.load()(self, ep.name, value)
        File "c:\users\gongxiaohui_2022\appdata\local\temp\pip-install-lnsomkd3\streamexpect_82c3615fd898415aa39c746558caadae\.eggs\setuptools_markdown-0.4.1-py3.9.egg\setuptools_markdown.py", line 43, in long_description_markdown_filename
          output = pypandoc.convert(markdown_filename, 'rst', format='md')
      AttributeError: module 'pypandoc' has no attribute 'convert'
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

�� Encountered error while generating package metadata.
�t��> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
CalledProcessError: Command '['C:\\Users\\gongxiaohui_2022\\.platformio\\penv\\Scripts\\python.exe', '-m', 'pip', 'install', 'streamexpect']' returned non-zero exit status 1.:
  File "C:\Users\gongxiaohui_2022\.platformio\penv\lib\site-packages\platformio\builder\main.py", line 196:
    env.SConscript(item, exports="env")
  File "C:\Users\gongxiaohui_2022\.platformio\packages\tool-scons\scons-local-4.4.0\SCons\Script\SConscript.py", line 597:
    return _SConscript(self.fs, *files, **subst_kw)
  File "C:\Users\gongxiaohui_2022\.platformio\packages\tool-scons\scons-local-4.4.0\SCons\Script\SConscript.py", line 285:
    exec(compile(scriptdata, scriptname, 'exec'), call_stack[-1].globals)
  File "C:\Users\gongxiaohui_2022\Desktop\ExpressLRS-2.2.0\src\python\build_env_setup.py", line 7:
    import ETXinitPassthrough
  File "C:\Users\gongxiaohui_2022\Desktop\ExpressLRS-2.2.0\src\python\ETXinitPassthrough.py", line 14:
    subprocess.check_call([sys.executable, "-m", "pip", "install", "streamexpect"])
  File "subprocess.py", line 373:
    
========================= [FAILED] Took 17.54 seconds =========================

Environment               Status    Duration
------------------------  --------  ------------
BETAFPV_2400_RX_via_UART  FAILED    00:00:17.540
==================== 1 failed, 0 succeeded in 00:00:17.540 ====================

**Comment by commanderguy3001** [comment]
it looks like the configurator is unable to install some required packages, it would help if you could tell us more about your computer so we can better help you.
Interesting would be things like:
- what operating system (windows / linux / macos / others) are you using?
- if you are in china, are you using a VPN while running the configurator?
- when did you install your operating system, and what other programs fo you have installed?

看起来配置器无法安装一些所需的软件包，如果你能告诉我们更多关于你的电脑的情况，我们就能更好地帮助你。
有意思的是像这样的事情。
- 你使用的是什么操作系统（windows / linux / macos / 其他）？
- 如果你在中国，你在运行配置器时是否使用VPN？
- 你是什么时候安装的操作系统，你还安装了什么其他程序？

**Comment by FPVLOVE** [comment]
I used Windows 10,and no used the VPN network, the PC system installed on September
kindly regards!
sunshine!

---- 回复的原邮件 ----
***@***.***>日期2022年11月26日 ***@***.***>***@***.***>***@***.***>主题Re: [ExpressLRS/ExpressLRS] expressLRS configurator 编译固件报错信息如何解决 (Discussion #1967)
it looks like the configurator is unable to install some required packages, it would help if you could tell us more about your computer so we can better help you.
Interesting would be things like:
what operating system (windows / linux / macos / others) are you using?
if you are in china, are you using a VPN while running the configurator?
when did you install your operating system, and what other programs fo you have installed?
看起来配置器无法安装一些所需的软件包，如果你能告诉我们更多关于你的电脑的情况，我们就能更好地帮助你。
有意思的是像这样的事情。
你使用的是什么操作系统（windows / linux / macos / 其他）？
如果你在中国，你在运行配置器时是否使用VPN？
你是什么时候安装的操作系统，你还安装了什么其他程序？

**Comment by jurgelenas** [comment]
ExpressLRS 2.X is no longer supported by the team. It also has issues with recent platformio releases.

Please upgrade to 3.x.

**Comment by FPVLOVE** [comment]
thanks for you reminds, okay,i will upgrade the firmware to 3.0.
kindly regards!
sunshine




| |
sunshine
***@***.***
|




---- 回复的原邮件 ----
| 发件人 | ***@***.***> |
| 日期 | 2022年11月28日 15:55 |
| 收件人 | ***@***.***> |
| 抄送至 | ***@***.***>***@***.***> |
| 主题 | Re: [ExpressLRS/ExpressLRS] expressLRS configurator 编译固件报错信息如何解决 (Discussion #1967) |

ExpressLRS 2.X is no longer supported by the team. It also has issues with recent platformio releases.

Please upgrade to 3.x.
